tests/unit/neuralnets/NeuralRectifiedLinearLayer_unittest.cc

```diff
@@ -31,47 +31,49 @@
  * Written (W) 2014 Khaled Nasr
  */
 
-#include <shogun/neuralnets/NeuralRectifiedLinearLayer.h>
-#include <shogun/neuralnets/NeuralInputLayer.h>
-#include <shogun/lib/SGVector.h>
-#include <shogun/lib/SGMatrix.h>
-#include <shogun/mathematics/Math.h>
-#include <gtest/gtest.h>
+#include "shogun/neuralnets/NeuralRectifiedLinearLayer.h"
+#include "shogun/lib/SGMatrix.h"
+#include "shogun/lib/SGVector.h"
+#include "shogun/mathematics/Math.h"
+#include "shogun/neuralnets/NeuralInputLayer.h"
+#include "utils/Utils.h"
+#include "gtest/gtest.h"
+
+#include <memory>
+#include <tuple>
 
 using namespace shogun;
 
+class NeuralRectifiedLinearLayerTest : public ::testing::Test
+{
+protected:
+	void SetUp()
+	{
+		CMath::init_random(100);
+		m_layers = std::make_unique<CDynamicObjectArray>();
+	}
+	std::unique_ptr<CDynamicObjectArray> m_layers;
+};
+
 /** Compares the activations computed using the layer against manually computed
  * activations
  */
-TEST(NeuralRectifiedLinearLayer, compute_activations)
+TEST_F(NeuralRectifiedLinearLayerTest, compute_activations)
 {
-	CNeuralRectifiedLinearLayer layer(9);
-
 	// initialize some random inputs
-	CMath::init_random(100);
-	SGMatrix<float64_t> x(12,3);
-	for (int32_t i=0; i<x.num_rows*x.num_cols; i++)
-		x[i] = CMath::random(-10.0,10.0);
-
-	CNeuralInputLayer* input = new CNeuralInputLayer (x.num_rows);
-	input->set_batch_size(x.num_cols);
-
-	CDynamicObjectArray* layers = new CDynamicObjectArray();
-	layers->append_element(input);
-
+	SGMatrix<float64_t> x;
+	CNeuralInputLayer* input;
+	std::tie(x, input) =
```

Comment:
I had a question about which C++ version is Shogun restricted to?

I can further trim this if C++17 is permitted.